esphomeyaml/components/time.rst

```diff
@@ -59,6 +59,35 @@ Configuration variables:
 - **on_time** (*Optional*, :ref:`Automation <automation>`): Automation to run at specific intervals using
   a cron-like syntax. See :ref:`time-on_time`.
 
+DS1307 Time Source
+------------------
+
+With the ``ds1307`` time platform, the battery powered :doc:`DS1307 <https://datasheets.maximintegrated.com/en/ds/DS1307.pdf>`
+clock chip connected via :doc:`I2C </esphomeyaml/components/i2c>` bus is used to synchronize the system time.
+
+.. code-block:: yaml
+
+    # Example configuration entry
+    time:
+      - platform: ds1307
+        set_interval: once
+        get_interval: 10min
+      # optionally add a time platform to sync the hardware clock to
+      - platform: sntp
+
+Configuration variables:
+
+- **id** (*Optional*, :ref:`config-id`): Specify the ID of the time for use in lambdas.
+- **on_time** (*Optional*, :ref:`Automation <automation>`): Automation to run
+  at specific intervals using a cron-like syntax. See :ref:`time-on_time`.
+- **set_interval** (*Optional*): Specify the interval at which the system time
+  is written to the hardware clock. By default, the hardware clock is only
+  written to once if the clock chip was not synchronized before and if the system
+  time is valid (i.e. synchronized from another time source).
```

Comment:
Potential problem with this "once" behavior will be that the "get" can interfere with it. For example on boot the DS1307 will be the first to get the current time - so then you would only be writing the time you got from the sensor back to it.

I would make it an interval with the magic value `never`. So for first setup users sets something like `1m` and after that can disable it.